verification/io/router_events.gobra

```diff
@@ -20,112 +20,112 @@ package io
+	return (d && hf1.InIF2 === some(recvif) && dp.valid_link_types_in2(hf1, asid)) ||
+		(!d && hf1.EgIF2 === some(recvif) && dp.valid_link_types2(hf1, asid))
 }
 
 ghost
 decreases
-pure func dp2_exit_interface(d bool, asid IO_as, hf1 IO_HF, outif IO_ifs) bool {
+pure func (dp DataPlaneSpec) dp2_exit_interface(d bool, asid IO_as, hf1 IO_HF, outif IO_ifs) bool {
 	return (d && hf1.EgIF2 == some(outif)) || (!d && hf1.InIF2 == some(outif))
 }
 
 // Takes a packet and forwards it externally, incrementing the segment before doing so.
 ghost
 decreases
-pure func dp2_forward_ext_guard(asid IO_as, m IO_pkt2, nextif IO_ifs, currseg, traversedseg IO_seg2, newpkt IO_pkt2, fut seq[IO_HF], hf1 IO_HF) bool {
+pure func (dp DataPlaneSpec) dp2_forward_ext_guard(asid IO_as, m IO_pkt2, nextif IO_ifs, currseg, traversedseg IO_seg2, newpkt IO_pkt2, fut seq[IO_HF], hf1 IO_HF) bool {
 	return m.CurrSeg == currseg &&
 		newpkt == IO_pkt2(IO_Packet2{traversedseg, m.LeftSeg, m.MidSeg, m.RightSeg}) &&
 		// The outgoing interface is correct:
-		dp2_exit_interface(currseg.ConsDir, asid, hf1, nextif) &&
+		dp.dp2_exit_interface(currseg.ConsDir, asid, hf1, nextif) &&
 		// Next validate the current hop field with the *original* UInfo field):
-		hf_valid(currseg.ConsDir, currseg.AInfo, currseg.UInfo, hf1) &&
+		dp.hf_valid(currseg.ConsDir, currseg.AInfo, currseg.UInfo, hf1) &&
 		hf1.extr_asid() == asid &&
 		// Segment update: push current hop field into the Past path and History:
-		inc_seg2(currseg, traversedseg, hf1, fut) &&
+		dp.inc_seg2(currseg, traversedseg, hf1, fut) &&
 		// If the current segment is an *down*-segment, then we need to update the
 		// uinfo field *after* validating the hop field
-		update_uinfo(currseg.ConsDir, currseg, traversedseg, hf1) &&
+		dp.update_uinfo(currseg.ConsDir, currseg, traversedseg, hf1) &&
 		// Other fields: no update
-		same_other2(currseg, traversedseg)
+		dp.same_other2(currseg, traversedseg)
 }
 
-ghost
-decreases
-pure func dp2_forward_ext(s IO_dp2_state, sprime IO_dp2_state, asid IO_as, m IO_pkt2) bool {
-	return exists nextif IO_ifs, currseg IO_seg2, traversedseg IO_seg2, newpkt IO_pkt2, fut seq[IO_HF], hf1 IO_HF ::  { dp2_forward_ext_guard(asid, m, nextif, currseg, traversedseg, newpkt, fut, hf1) }{ dp2_add_ext(s, sprime, asid, nextif, newpkt) } dp2_forward_ext_guard(asid, m, nextif, currseg, traversedseg, newpkt, fut, hf1) &&
-		dp2_add_ext(s, sprime, asid, nextif, newpkt)
+// ghost
+// decreases
+// pure func dp2_forward_ext(s IO_dp2_state, sprime IO_dp2_state, asid IO_as, m IO_pkt2) bool {
+// 	return exists nextif IO_ifs, currseg IO_seg2, traversedseg IO_seg2, newpkt IO_pkt2, fut seq[IO_HF], hf1 IO_HF ::  { dp2_forward_ext_guard(asid, m, nextif, currseg, traversedseg, newpkt, fut, hf1) }{ dp2_add_ext(s, sprime, asid, nextif, newpkt) } dp2_forward_ext_guard(asid, m, nextif, currseg, traversedseg, newpkt, fut, hf1) &&
+// 		dp2_add_ext(s, sprime, asid, nextif, newpkt)
 
-}
+// }
```

Comment:
Again, I am sorry for the change of heart; if these are really not used, let's drop them.